core/src/com/unciv/ui/newgamescreen/NewGameScreen.kt

```diff
@@ -100,13 +101,29 @@ class NewGameScreen(private val previousScreen: CameraStageBaseScreen, _gameSetu
                 if (rulesetIncompatibilities.isNotEmpty()) {
                     val incompatibleMap = Popup(this)
                     incompatibleMap.addGoodSizedLabel("Map is incompatible with the chosen ruleset!".tr()).row()
-                    for(incompat in rulesetIncompatibilities)
-                        incompatibleMap.addGoodSizedLabel(incompat).row()
+                    for(incompatibility in rulesetIncompatibilities)
+                        incompatibleMap.addGoodSizedLabel(incompatibility).row()
                     incompatibleMap.addCloseButton()
                     incompatibleMap.open()
                     game.setScreen(this) // to get the input back
                     return@onClick
                 }
+            } else {
+                // Generated map - check for sensible dimensions and if exceeded correct them and notify user
+                val mapSize = gameSetupInfo.mapParameters.mapSize
+                val message = mapSize.fixUndesiredSizes(gameSetupInfo.mapParameters.worldWrap)
+                if (message != null) {
+                    Gdx.app.postRunnable {
+                        ToastPopup( message, UncivGame.Current.screen as CameraStageBaseScreen, 4000 )
+                        with (mapOptionsTable.generatedMapOptionsTable) {
```

Comment:
Why do we have this separately in both screens? This sounds like something that should go directly in the MapParametersTable that's shared by both screens

Reply:
True, and the idea occurred to me, but `MapParametersTable` isn't in the `ui` top level container, it's `logic`... and having the 'writing back of corrected values to the widgets' abstract enough so they could share that portion of code - too much refactor for my taste. I _was_ tempted to reorganize those screens from scratch... Major project though.

Reply:
I think this has now become necessary, but I see your point. This fix could go in a different PR.